DataContent/DAL/Repositories/UserRepository.cs

```diff
@@ -96,9 +97,23 @@ public async Task<User> UpdateUserAsync(User user)
                 updatedUser.Email = user.Email;
                 updatedUser.Username = user.Username;
                 updatedUser.Password = new ScryptEncoder().Encode(user.Password);
+                updatedUser.RecoveryPassword = null;
             }
             await _context.SaveChangesAsync();
             return updatedUser;
         }
+
+        public async Task<User> ForgotPasswordAsync(string email)
+        {
+            var user = await _context.Users.Where(x => x.Email == email).FirstOrDefaultAsync();
+            if(user.RecoveryDate.AddSeconds(30) < DateTime.Now)
+            {
+                user.RecoveryPassword = PasswordHelper.RandomString(16);
+                user.RecoveryDate = DateTime.Now;
+                await _context.SaveChangesAsync();
+            }
+            
+            return user;
```

Comment:
should throw exception if recoverydate was less than 30s, so we could show an error to user later

Reply:
Done